### PR TITLE
injector: remove canonical service/revision env variables

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

It would be better to have envoy perform the computation but that requires a new feature and can be done in a follow up.

### CODE REVIEW

**Comment by howardjohn** [comment]
LGTM, will let someone familiar with tracing code approve as well though

File: pilot/pkg/networking/core/v1alpha3/tracing.go [hunk 1]
```diff
@@ -350,39 +350,59 @@ func defaultTags() []*tracing.CustomTag {
...
+	var revision, service string
+	if metadata.Labels != nil {
+		revision = metadata.Labels["service.istio.io/canonical-revision"]
+		service = metadata.Labels["service.istio.io/canonical-name"]
+	}
+	if revision == "" {
+		revision = "latest"
+	}
+	if service == "" {
+		service = "unknown"
```
Comment (douglas-reid) [inline pilot/pkg/networking/core/v1alpha3/tracing.go:366]: this makes me a bit uneasy. it is probably overkill to rerun the fallback for other labels though.

Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/tracing.go:366]: This is preserving the original behavior. What would you prefer here (we have more options now)?

File: pilot/pkg/networking/core/v1alpha3/tracing.go [hunk 2]
```diff
@@ -435,6 +455,7 @@ func configureCustomTags(hcmTracing *hpb.HttpConnectionManager_Tracing,
 	if features.EnableIstioTags {
 		defaultTags := defaultTags()
 		tags = append(tags, defaultTags...)
+		tags = append(tags, buildCustomTracingTags(metadata)...)
```
Comment (douglas-reid) [inline pilot/pkg/networking/core/v1alpha3/tracing.go:458]: nit: should we update `defaultTags()` to be `optionalPolicyTags()` (or similar) and rename `buildCustomTracingTags()` to `buildServiceTags()` (or similar) to make what is happening clearer?

File: pilot/pkg/networking/core/v1alpha3/tracing.go (hunk 1 above)
Comment (douglas-reid) [reply pilot/pkg/networking/core/v1alpha3/tracing.go:366]: I think this is actually fine. I was just thinking out loud - sorry about that. 

Maybe we can add a comment here to say "this should have been properly handled during proxy injection." ?

Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/tracing.go:366]: Maybe file an issue with the concern and I'll link it here?

File: pilot/pkg/networking/core/v1alpha3/tracing.go (hunk 2 above)
Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/tracing.go:458]: Yeah, good idea. Default is not appropriate here.

Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/tracing.go:458]: Done

File: pilot/pkg/networking/core/v1alpha3/tracing.go (hunk 1 above)
Comment (kyessenov) [reply pilot/pkg/networking/core/v1alpha3/tracing.go:366]: Added a comment.

**Review by douglas-reid — APPROVED** [review]

**Comment by kyessenov** [comment]
@howardjohn Do you mind approving?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**